### PR TITLE
Added getTagValue to ZMQ tx topic

### PR DESCRIPTION
**Pull request by SteveFromTheOffice**
It's about time, anyways!

### CODE REVIEW

**Comment by th0br0** [comment]
Don't replace the field, please.
I'm happy to accept a PR to append the tag to the ZMQ output as necessary though. (in the right position of course)

**Comment by cwarner818** [comment]
Why would you want to keep `obsoleteTag`? Isn't it obsolete?

**Comment by SteveFromTheOffice** [comment]
Appended it instead, based it on the tryte order (what I assumed was the right position). If not, let me know!

**Comment by cwarner818** [comment]
It doesn't hold any data AFAIK. It's just a copy of the `tag` that gets overwritten during hashing. Am I misunderstanding it?

**Comment by cwarner818** [comment]
What if we expanded the transaction output via zeromq to contain all transaction fields, and change as necessary as IRI evolves? That way we include all the data and its up to the foundation to not keep switching up the protocol :) 

But for now I think it makes a lot of sense to switch out `obosleteTag` for `tag` unless someone has a specific use case for keeping `obosleteTag`.

File: src/main/java/com/iota/iri/storage/ZmqPublishProvider.java [hunk 1]
```diff
@@ -148,4 +149,4 @@ public void clearMetadata(Class<?> column) throws Exception {
 
     }
 
-}
\ No newline at end of file
+}
```
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/storage/ZmqPublishProvider.java]: Why did you add an extra line?

**Comment by GalRogozinski** [comment]
Please append the new value at the very end. This is so that people who parse the ZMQ tx topic according to positions won't be affected by movement ArrivalTime's position.

Also, maybe you are willing to edit `com/iota/iri/zmq/README.md` to describe the ZMQ topic structure.

**Comment by cwarner818** [comment]
Isn't obsoleteTag going to be removed entirely in the future? Why not rip that bandaid off now instead of pushing it off until later? Everything I've seen that uses obsoleteTag is just using it to display the tag field as if it was still valid (ie http://tangle.glumb.de/) anyways.

**Comment by SteveFromTheOffice** [comment]
Apologies for the delay, was MIA for a few weeks. I'm also willing to look at the readme and add details to it, but I think we should PR that separately.

**Comment by GalRogozinski** [comment]
@SteveFromTheOffice If you can open a separate PR like you suggested it will be great!

**Comment by alon-e** [comment]
This PR is ready for merging, but requires informing ZMQ users,
even though ZMQ is still experimental and non-default - there is code dependant on the format.
@geminoz @viossat @alon-e @tsvi -- please test this PR against your existing implementations and report back.

**Comment by viossat** [comment]
@alon-e It's ok here. (thanks for the ping)
Please also consider #447

**Comment by geminoz** [comment]
Saw this now. 

Heaps of changes in the dev branch in the past week. Is a rebase of this PR required to test against the newer code?

**Comment by alon-e** [comment]
@gem for testing your infra no, just run this PR and test the `tx` topic parser.

But in order to merge it, a small rework is needed, to match #447 merged changes.

**Comment by tsvisabo** [comment]
No problems on my side

**Comment by alon-e** [comment]
works on my implementation also. 
-- @SteveFromTheOffice while reworking PR to match latest changes the PR lost context to upstream, hence I lost the ability to push the rework to your fork. sorry about that.

I'll create a new PR with the correct changes and merge.
